### ISSUE TITLE
Error while publishing branch on a new repo

### ISSUE THREAD

**Issue by oyeanuj** [desktop/desktop#3086]
### Description

This issue was originally discussed in #1286 and I can still reproduce it. I created a new repo on github.com and after creating it, tried both cloning it via Github Desktop and using 'opening repo in Desktop' option. In both cases, the repo is cloned and the app shows me an option to 'Publish branch'. When I click on it, nothing happens in the UI. On the console, I see the following promise error (also seen in #1286).

```
Uncaught (in promise) Error: The current branch is unborn.
```

### Version
1.0.3

**GitHub Desktop version:** 1.0.3

**OS version:** macOS High Sierra 10.13

### Steps to Reproduce

1. Create a new repo on Github.com
2. Clone that repo (either via Github Desktop) or via Github.com (Open in Desktop option).
3. Notice the enabled Publish button.
4. Click on it.
5. Nothing happens in the UI.
6. Open console, notice the error.

**Expected behavior:** Default branch to be published/synced.

**Actual behavior:** Nothing. Errors out but doesn't inform user.

**Reproduces how often:** 100%

**Comment by shiftkey** [comment]
@oyeanuj I think the last incidence of this was in `1.0.3` - https://github.com/desktop/desktop/issues/2773#issuecomment-334160456 

Can you retest on `1.0.4`? This PR changes how we enable/disable the publish button: https://github.com/desktop/desktop/pull/2966

**Comment by oyeanuj** [comment]
@shiftkey You are right, the button does seem disabled now. Thank you!

If I may suggest, a more descriptive error message would help people figure out the next step. The current copy is technically accurate but doesn't indicate what needs to be done next or what is happening in layman terms.

Closing this ticket since the original issue seems to be taken care of and don't want to overload the issues list!
